sdk/core/System.ClientModel/tests/TestFramework/Mocks/MockPipelineResponse.cs

```diff
@@ -73,4 +101,48 @@ protected void Dispose(bool disposing)
             _disposed = true;
         }
     }
+
+    public override BinaryData ReadContent(CancellationToken cancellationToken = default)
+    {
+        if (_bufferedContent is not null)
+        {
+            return _bufferedContent;
+        }
+
+        if (_contentStream is null)
+        {
+            _bufferedContent = BinaryData.FromString(string.Empty);
+            return _bufferedContent;
+        }
+
+        MemoryStream bufferStream = new();
+        _contentStream.CopyTo(bufferStream);
+        _contentStream.Dispose();
+        _contentStream = bufferStream;
+
+        _bufferedContent = BinaryData.FromStream(bufferStream);
+        return _bufferedContent;
+    }
+
+    public override async ValueTask<BinaryData> ReadContentAsync(CancellationToken cancellationToken = default)
+    {
+        if (_bufferedContent is not null)
+        {
+            return _bufferedContent;
+        }
+
+        if (_contentStream is null)
+        {
+            _bufferedContent = BinaryData.FromString(string.Empty);
+            return _bufferedContent;
+        }
+
+        MemoryStream bufferStream = new();
+        await _contentStream.CopyToAsync(bufferStream).ConfigureAwait(false);
+        _contentStream.Dispose();
```

Comment:
can we call DisposeAsync in async methods?

Reply:
Addressed